src/push.rs

```diff
@@ -99,12 +99,46 @@ pub async fn push_profile(data: PushProfileData<'_>) -> Result<(), PushProfileEr
         Command::new("nix-build")
     };
 
+    let store_address = format!("ssh-ng://{}@{}",
+        if data.deploy_data.profile.generic_settings.ssh_user.is_some() {
+            &data.deploy_data.profile.generic_settings.ssh_user.as_ref().unwrap()
+        } else {
+            &data.deploy_defs.ssh_user
+        },
+        data.deploy_data.node.node_settings.hostname
+    );
+
     if data.supports_flakes {
         build_command.arg("build").arg(derivation_name)
     } else {
         build_command.arg(derivation_name)
     };
 
+    // copy derivation over to remote server
+    if data.supports_flakes && data.deploy_data.merged_settings.remote_build.unwrap_or(false) {
+        info!("Building on remote host");
+
+        // copy the derivation to remote host so it can be built there
+        let copy_command_status = Command::new("nix").arg("copy")
+            .arg("-s")  // fetch dependencies from substitures, not localhost
+            .arg("--to").arg(&store_address)
+            .arg("--derivation").arg(derivation_name)
```

Comment:
Something like
```suggestion
            .arg("--derivation").arg(derivation_name)
            .env("NIX_SSHOPTS", &ssh_opts_str)
```
And moving `ssh_opts_str` definition above seems to help

Reply:
Hmm, after this change deployment hangs in
```
🚀 ℹ️ [deploy] [INFO] Building on remote host
```
state. And I don't see `nix build` on a target machine :(